### ISSUE TITLE
Downloading a RAR-compressed patch causes the Local Storage table to break

### ISSUE THREAD

**Issue by meanmedianmoge** [meanmedianmoge/zoia_lib#80]
**Describe the bug**
Local Storage table is empty after I downloaded a patch that is saved as a .rar file. Thanks to Jason for submitting this bug through the FB group. Confirmed on my machine.

**To Reproduce**
1. Search for "Zoia Murf" in the PS tab, download v2.0 (1.0 is a bin file)
2. Move to LS tab, notice how the table doesn't populate
3. Investigate the /.ZoiaLibraryApp backend, see a /Temp folder
4. See terminal error:
```
Traceback (most recent call last):
  File "/Users/rkm345/PycharmProjects/personal/zoia_lib/zoia_lib/UI/ZOIALibrarian_main.py", line 358, in tab_switch
    self.local.get_local_patches()
  File "/Users/rkm345/PycharmProjects/personal/zoia_lib/zoia_lib/UI/ZOIALibrarian_local.py", line 130, in get_local_patches
    for pch in os.listdir(os.path.join(self.path, patches)):
NotADirectoryError: [Errno 20] Not a directory: '/Users/rkm345/Library/Application Support/.ZoiaLibraryApp/temp.zip'
```

**Expected behavior**
Expected to be able to download a patch and have it show in the Local Storage table. Having the rar format seems to break things.

**Screenshot**
![image](https://user-images.githubusercontent.com/51132973/94489827-354dd880-01b3-11eb-861f-754a73c3b314.png)

**Desktop (please complete the following information):**
- Mac OS X 10.15.4

**Additional context**
ZoiaMurf patch is saved as a .rar file, which requires a special de-compression method. Using Unarchiver (Mac unzipper program), this causes issues in the backend.. see a folder labelled "Temp". This is unexpected. Deleting the directory from the backend fixes the issue. Also, uploading the patch as a zip would avoid the issue.

**Comment by meanmedianmoge** [comment]
```
-----------------------------------------------------------
BadRarFile                Traceback (most recent call last)
<ipython-input-45-cb3c4ed70912> in <module>
----> 1 rar_obj.extractall(pch)

/usr/local/lib/python3.8/site-packages/rarfile.py in extractall(self, path, members, pwd)
    845         for m in members:
    846             inf = self.getinfo(m)
--> 847             dst = self._extract_one(inf, path, pwd, not inf.is_dir())
    848             if inf.is_dir():
    849                 if dst not in done:

/usr/local/lib/python3.8/site-packages/rarfile.py in _extract_one(self, info, path, pwd, set_attrs)
    910 
    911         if info.is_file():
--> 912             return self._make_file(info, dstfn, pwd, set_attrs)
    913         if info.is_dir():
    914             return self._make_dir(info, dstfn, pwd, set_attrs)

/usr/local/lib/python3.8/site-packages/rarfile.py in _make_file(self, info, dstfn, pwd, set_attrs)
    925         with self.open(info, "r", pwd) as src:
    926             with open(dstfn, "wb", opener=helper) as dst:
--> 927                 shutil.copyfileobj(src, dst)
    928         if set_attrs:
    929             self._set_attrs(info, dstfn)

/usr/local/Cellar/python@3.8/3.8.4/Frameworks/Python.framework/Versions/3.8/lib/python3.8/shutil.py in copyfileobj(fsrc, fdst, length)
    200     fdst_write = fdst.write
    201     while True:
--> 202         buf = fsrc_read(length)
    203         if not buf:
    204             break

/usr/local/lib/python3.8/site-packages/rarfile.py in read(self, n)
   2195         data = b"".join(buf)
   2196         if n > 0:
-> 2197             raise BadRarFile("Failed the read enough data: req=%d got=%d" % (orig, len(data)))
   2198 
   2199         # done?

BadRarFile: Failed the read enough data: req=32768 got=51
```

**Comment by smithrobs** [comment]
You probably don't want to hear this but 7-zip decompresses https://patchstorage.com/zoia-murf-v2-0/ with no errors. ☹️

**Comment by Arxi** [comment]
I've just come across this issue after downloading Zoia Murf V2 - is there any intermittent quick fix I can do? Some files I can delete manually, for example? Thank you!

**Comment by meanmedianmoge** [comment]
@Arxi yes, there is an easy remedy for this. See the issue I just mentioned (80) for a bit more explanation.

In short, you’ll have to delete a file or two and things will go back to normal (you’re looking for a Temp directory and temp.zip I believe). I blame the developers of RAR for making a compression algo that requires a license.

Windows: `C:\Users\user\AppData\Roaming\.ZoiaLibraryApp\`
Mac: `/Users/user/.Library/Application` Support/.ZoiaLibraryApp/
Linux: `/home/.local/share/.ZoiaLibraryApp/`

**Comment by Arxi** [comment]
@meanmedianmoge thank you! In the end, I had to delete (on OSX):
`~/.Library/Application Support/.ZoiaLibraryApp/temp.zip`
and also a .rar file of Zoia Murf V2 in one of the 6-digit directories I believe. So just as you said!

**Comment by meanmedianmoge** [comment]
Forgot to close this. This patch for this issue will be included in the v1.1 release!
